admin/orders.php

```diff
@@ -287,12 +312,20 @@
           }
         }
         $messageStack->add_session(SUCCESS_ORDER_UPDATED, 'success');
+          if ($customer_notified === 1) {
+              $messageStack->add_session(sprintf(SUCCESS_EMAIL_SENT, ($admin_language !== $_SESSION['languages_code'] ? '(' . strtoupper($_SESSION['languages_code']) . ') ' : '')), 'success'); // show an email sent confirmation message, with a language indicator if the order/email language was different to the admin user language
+          }
         zen_record_admin_activity('Order ' . $oID . ' updated.', 'info');
       } else {
         $messageStack->add_session(WARNING_ORDER_NOT_UPDATED, 'warning');
       }
-      zen_redirect(zen_href_link(FILENAME_ORDERS, zen_get_all_get_params(array('action')) . 'action=edit', 'NONSSL'));
+
+        if (isset($_POST['camefrom']) && $_POST['camefrom'] === 'orderEdit') {
+            zen_redirect(zen_href_link(FILENAME_ORDERS, zen_get_all_get_params(['action', 'language']) . 'action=edit' . ($admin_language !== $_SESSION['languages_code'] ? '&language=' . $admin_language : ''), 'NONSSL'));
+        }
+        zen_redirect(zen_href_link(FILENAME_ORDERS, zen_get_all_get_params(['action', 'language']) . ($admin_language !== $_SESSION['languages_code'] ? 'language=' . $admin_language : ''), 'NONSSL'));
```

Comment:
What do you think of the following approach, which makes the `if` intention slightly clearer (ie: that the only difference is the `edit` action)?

```php
        $redirect = zen_href_link(FILENAME_ORDERS, zen_get_all_get_params(['action', 'language']) . ($admin_language !== $_SESSION['languages_code'] ? '&language=' . $admin_language : ''), 'NONSSL');
        if (isset($_POST['camefrom']) && $_POST['camefrom'] === 'orderEdit') {
            $redirect .= '&action=edit';
         }
         zen_redirect($redirect);
```

Reply:
ok.